src/apps/ocioconvert/main.cpp

```diff
@@ -422,7 +421,11 @@ int main(int argc, const char **argv)
 
         try
         {
-            oglApp = std::make_shared<OCIO::OglApp>("ocioconvert", 256, 20);
+# ifdef OCIO_HEADLESS_ENABLED
+            oglApp = std::make_shared<OCIO::HeadlessApp>("ocioconvert", 256, 20);
+# else
+            oglApp = std::make_shared<OCIO::ScreenApp>("ocioconvert", 256, 20);
+# endif
```

Comment:
The code should have a factory method (i.e. a static method to create the right class instance) attached to the base class. It simplifies the code everywhere headless or screen are created i.e. mainly main.cpp of several apps.